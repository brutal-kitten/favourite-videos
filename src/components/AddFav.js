import React, { Component } from 'react';
import PropTypes from 'prop-types';

class AddFav extends Component {

  static propTypes = {
    addToFavorite: PropTypes.func.isRequired,
    trigger: PropTypes.func.isRequired,
    id: PropTypes.string.isRequired
  }

  //calls function to add item to favorites and trigger the change of button
  handleClick = (event) => {

    event.preventDefault();
    this.props.addToFavorite(this.props.id);
    this.props.trigger();

  }


  render() {

    return(
      <div className="addFav" >
        <button tabIndex='0' className="buttonAddFav" type="button" onClick={(event) => this.handleClick(event)}>
          <span className="glyphicon glyphicon-star-empty"></span></button>
      </div>
    )
  }
}

export default AddFav;
