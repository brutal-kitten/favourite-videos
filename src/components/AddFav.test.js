import React from 'react';
import ReactDOM from 'react-dom';
import AddFav from './AddFav.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


it('should display a button', () => {
  const tree = TestRenderer.create(
    <AddFav />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe ('shoud call handleclick function after click', () => {
  const mockAddToFavorite = jest.fn();
  const mockTrigger = jest.fn();
  const props = {
    addToFavorite: mockAddToFavorite,
    trigger: mockTrigger,
    id: "12345"
  };

  const wrapper = shallow(<AddFav {...props}/>);

  wrapper.find('button').simulate('click', {
    preventDefault: () => {},
  });

  it('should call addToFavorite function with 1 argument - id', () => {
    expect(mockAddToFavorite.mock.calls.length).toBe(1);
    expect(mockAddToFavorite.mock.calls[0][0]).toBe("12345");
  });

  it('should call trigger function', () => {
    expect(mockTrigger.mock.calls.length).toBe(1);
  });

})
