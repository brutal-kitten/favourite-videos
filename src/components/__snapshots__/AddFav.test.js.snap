// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a button 1`] = `
<div
  className="addFav"
>
  <button
    className="buttonAddFav"
    onClick={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="glyphicon glyphicon-star-empty"
    />
  </button>
</div>
`;
